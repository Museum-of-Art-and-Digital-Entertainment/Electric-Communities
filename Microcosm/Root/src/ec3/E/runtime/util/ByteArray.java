/*
	byte[] utilities
*/

package ec.util;

public class ByteArray {

public static byte[] concat(byte[] a, byte[] b) {
    byte[] c = new byte[a.length + b.length];
    for(int i = 0; i < a.length; i++) {
      c[i] = a[i];
    }
    for(int i = 0; i < b.length; i++) {
      c[a.length + i] = b[i];
    }
    return c;
  }

}
