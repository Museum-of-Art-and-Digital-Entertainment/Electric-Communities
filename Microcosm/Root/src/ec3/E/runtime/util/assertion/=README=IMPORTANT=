This is a COPY of the code in evm1/ec/util/assertion.  Be careful not
to change this.

The makefile is different.
