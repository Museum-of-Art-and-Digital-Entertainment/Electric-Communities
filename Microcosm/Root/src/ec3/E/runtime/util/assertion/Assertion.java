/*
 *  Java/E assertions.  Written by Brian Marick,
 *  June 1997, for Electric Communities, Inc.
 *  Copyright (c) 1997 by Electric Communities.  All Rights Reserved.
 */


package ec.util.assertion;

/* DANGER                                                       DANGER

   The 'unassert' program will remove any invocation of ANY method
   of this class.  If you add a new non-private method, it MUST be
   one which can be removed without changing the semantics of the 
   invoking program.

   There has been talk of adding an "assertNoException" assertion that
   would act as a try...catch around the method.  That is an assertion
   the unassert program should not remove.  If that method is added,
   the code in ec.transform.assertion.UnAsserter#isAssertionToRemove
   will have to be changed.

   The unassert program also assumes that all assertions are static,
   and that they do not return a value.

   DANGER                                                       DANGER
*/

/** 
 * This class provides assertions you can add to your program.  If the
 * assertions fail, unchecked exceptions of class AssertionFailed are
 * thrown. 
 * <p>
 * Note:  an assertion that is true (does not throw) nevertheless
 * incurs the overhead of the method invocation.
 * Assertions can be removed
 * from the class file by a transforming class loader or a program
 * that rewrites the class file, as described in
 * <a href="stripping.html">Ways to Strip Assertions</a>.
 * That page also describes some 'gotchas' to beware when using
 * assertions. 
 * @see AssertionFailed
 */
public class Assertion
{
  static private final String prefix = "Internal program error: ";

  /* 
   * This variable is used to discourage a java compiler from inlining 
   * an assertion.  An inlined assertion would prevent the unasserter
   * from working.
   * I can't offhand think of a way to *prevent* inlining.
   */
  static private boolean preventInlining = false;

  /**
   * If the argument is false, throws an AssertionFailed with the message
   * "Internal program error: Assertion failed."
   */
  public static void test(boolean mustBeTrue) throws AssertionFailed {
    if (mustBeTrue == false) {
      throw new AssertionFailed(prefix + "Assertion failed.");
    }
    if (preventInlining) test(!mustBeTrue);
  }

  /**
   * If the argument is false, throws an AssertionFailed with the
   * given explanation, prefixed by "Internal program error: ".
   */
  public static void test(boolean mustBeTrue, String explanation)
       throws AssertionFailed {
    if (mustBeTrue == false) {
      throw new AssertionFailed(prefix + explanation);
    }
    if (preventInlining) test(!mustBeTrue, explanation);
  }

  /**
   * If the argument is false, throws an AssertionFailed with the
   * given explanation arguments, concatenated as strings and prefixed
   * by "Internal program error: ".
   * <p> Use this routine when you need to avoid paying the overhead of
   * string concatenation ("+") on every test.  It does the concatenation
   * only if the test fails.
   */
  public static void test(boolean mustBeTrue, Object explanation0,
                          Object explanation1)
       throws AssertionFailed {
    if (mustBeTrue == false) {
      throw new AssertionFailed(prefix + explanation0 + explanation1);
    }
    if (preventInlining) test(!mustBeTrue);
  }

  /**
   * If the argument is false, throws an AssertionFailed with the
   * given explanation arguments, concatenated as strings and prefixed
   * by "Internal program error: ".
   * <p> Use this routine when you need to avoid paying the overhead of
   * string concatenation ("+") on every test.  It does the concatenation
   * only if the test fails.
   */
  public static void test(boolean mustBeTrue, Object explanation0,
                          Object explanation1, Object explanation2)
       throws AssertionFailed {
    if (mustBeTrue == false) {
      throw new AssertionFailed(prefix + explanation0 +
                                explanation1 + explanation2);
    }
    if (preventInlining) test(!mustBeTrue);
  }

  /**
   * If the argument is false, throws an AssertionFailed with the
   * given explanation arguments, concatenated as strings and prefixed
   * by "Internal program error: ".
   * <p> Use this routine when you need to avoid paying the overhead of
   * string concatenation ("+") on every test.  It does the concatenation
   * only if the test fails.
   */
  public static void test(boolean mustBeTrue, Object explanation0,
                          Object explanation1, Object explanation2,
                          Object explanation3)
       throws AssertionFailed {
    if (mustBeTrue == false) {
      throw new AssertionFailed(prefix + explanation0 +
                                explanation1 + explanation2 +
                                explanation3);
    }
    if (preventInlining) test(!mustBeTrue);
  }

  /**
   * If the argument is false, throws an AssertionFailed with the
   * given explanation arguments, concatenated as strings and prefixed
   * by "Internal program error: ".
   * <p> Use this routine when you need to avoid paying the overhead of
   * string concatenation ("+") on every test.  It does the concatenation
   * only if the test fails.
   */
  public static void test(boolean mustBeTrue, Object explanation0,
                          Object explanation1, Object explanation2,
                          Object explanation3, Object explanation4)
       throws AssertionFailed {
    if (mustBeTrue == false) {
      throw new AssertionFailed(prefix + explanation0 +
                                explanation1 + explanation2 +
                                explanation3 + explanation4);
    }
    if (preventInlining) test(!mustBeTrue);
  }


  /**
   * If the argument is false, throws an AssertionFailed with the
   * given explanation arguments, concatenated as strings and prefixed
   * by "Internal program error: ".
   * <p> Use this routine when you need to avoid paying the overhead of
   * string concatenation ("+") on every test.  It does the concatenation
   * only if the test fails.
   */
  public static void test(boolean mustBeTrue, Object explanation0,
                          Object explanation1, Object explanation2,
                          Object explanation3, Object explanation4,
                          Object explanation5)
       throws AssertionFailed {
    if (mustBeTrue == false) {
      throw new AssertionFailed(prefix + explanation0 +
                                explanation1 + explanation2 +
                                explanation3 + explanation4 +
                                explanation5);
    }
    if (preventInlining) test(!mustBeTrue);
  }

  /**
   * If the argument is false, throws an AssertionFailed with the
   * given explanation arguments, concatenated as strings and prefixed
   * by "Internal program error: ".
   * <p> Use this routine when you need to avoid paying the overhead of
   * string concatenation ("+") on every test.  It does the concatenation
   * only if the test fails.
   */
  public static void test(boolean mustBeTrue, Object explanation0,
                          Object explanation1, Object explanation2,
                          Object explanation3, Object explanation4,
                          Object explanation5, Object explanation6)
       throws AssertionFailed {
    if (mustBeTrue == false) {
      throw new AssertionFailed(prefix + explanation0 +
                                explanation1 + explanation2 +
                                explanation3 + explanation4 +
                                explanation5 + explanation6);
    }
    if (preventInlining) test(!mustBeTrue);
  }

  /**
   * If the argument is false, throws an AssertionFailed with the
   * given explanation arguments, concatenated as strings and prefixed
   * by "Internal program error: ".
   * <p> Use this routine when you need to avoid paying the overhead of
   * string concatenation ("+") on every test.  It does the concatenation
   * only if the test fails.
   */
  public static void test(boolean mustBeTrue, Object explanation0,
                          Object explanation1, Object explanation2,
                          Object explanation3, Object explanation4,
                          Object explanation5, Object explanation6,
                          Object explanation7)
       throws AssertionFailed {
    if (mustBeTrue == false) {
      throw new AssertionFailed(prefix + explanation0 +
                                explanation1 + explanation2 +
                                explanation3 + explanation4 +
                                explanation5 + explanation6 +
                                explanation7);
    }
    if (preventInlining) test(!mustBeTrue);
  }

  /**
   * If this method is executed, it throws an AssertionFailed with the
   * message "Internal program error:  'Unreachable' code was reached."  Plant
   * such assertions in places the program should never reach (such as
   * the default case in a switch).
   */
  public static void fail() throws AssertionFailed {
    if (preventInlining) fail();
    throw new AssertionFailed(prefix + "'Unreachable' code was reached.");
  }

  /**
   * If this method is executed, it throws an AssertionFailed with the
   * given explanation, prefixed by "Internal program error: ".
   */
  public static void fail(String explanation) throws AssertionFailed {
    if (preventInlining) fail(explanation);
    throw new AssertionFailed(prefix + explanation);
  }
}
