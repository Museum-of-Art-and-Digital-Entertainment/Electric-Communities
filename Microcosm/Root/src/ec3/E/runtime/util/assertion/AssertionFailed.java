/*
 *  Java/E assertions.  Written by Brian Marick,
 *  June 1997, for Electric Communities, Inc.
 *  Copyright (c) 1997 by Electric Communities.  All Rights Reserved.
 */
package ec.util.assertion;

/**
 * Instances of this class are thrown by failing assertions.
 * @see Assertion
 */
public class AssertionFailed extends Error {
  public AssertionFailed() {
    super();
  }
  public AssertionFailed (String msg) {
    super(msg);
  }
}

