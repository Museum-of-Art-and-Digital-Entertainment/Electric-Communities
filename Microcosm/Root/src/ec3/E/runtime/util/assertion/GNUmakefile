PACKAGE = ec.util.assertion
MODNAME = util

FILES.javac = \
	Assertion.java \
	AssertionFailed.java

include $(TOP)/E/makefiles/Defs.gmk

all: mk_dirs classes

mk_dirs:
	mkdir -p $(FULLPATH)
	mkdir -p $(LISTPATH)

clean: basic_clean

include $(TOP)/E/makefiles/Rules.gmk

