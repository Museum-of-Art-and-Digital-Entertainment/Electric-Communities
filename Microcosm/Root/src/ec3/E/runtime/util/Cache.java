/*
    Cache
*/

package ec.util;

import java.util.Hashtable;
import java.util.Vector;
import java.util.Enumeration;


public class Cache {

  private Hashtable h, nodetable;
  public List l;
  public int size, maxsize;

  public Cache(int s) {       // s == maximum number of elements in cache
     size = 0;
     maxsize = s;
     h = new Hashtable(maxsize);
     nodetable = new Hashtable(maxsize);
     l = new List();
  }

  public Object get(Object key) {

    // bring this element to tail of list (== Most Recently Used)
    ListNode ln = (ListNode)nodetable.get(key);
    if (ln != null && ln != l.tail) {
      l.delete((ListNode)nodetable.get(key));
      l.add(key);
      nodetable.put(key, (Object)l.tail);
    }

    return h.get(key);
  }

  public Object put(Object key, Object data) {

    Object deleted_object = null;

    // add data to cache
    if (h.put(key, data) != null) {
      // just over-wrote something in the cache. 
      // remove the original entry from the linked list.
      l.delete((ListNode)nodetable.get(key));
    } else {
      size++;
    } 
    l.add(key);
    nodetable.put(key, l.tail);

    // have we reached cache capacity?  
    // Get rid of head element (== least recently used)
    if (size > maxsize) {
      deleted_object = h.get(l.head.data);
      h.remove(l.head.data);
      nodetable.remove(l.head.data);
      l.delete(l.head);
      size--;
    }

    return deleted_object;

  }

  public void flush() {
     h = new Hashtable(maxsize);
     nodetable = new Hashtable(maxsize);
     l = new List();
  }

  public Enumeration dump() { // for debugging, return enum of cache contents
    Vector v = new Vector();
    ListNode ln = l.head;
    while (ln != null){
      v.addElement(h.get(ln.data));
      ln = ln.next;
      }
    return v.elements();
  }
     
}
