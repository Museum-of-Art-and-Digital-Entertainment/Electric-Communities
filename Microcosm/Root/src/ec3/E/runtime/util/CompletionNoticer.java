package ec.util;

import java.util.Hashtable;
import java.util.Enumeration;

/**
   Target for callback when an operation is completed.
*/
public interface CompletionNoticer 
{
    void noticeCompletion(Object arg);
}

/**
   Generate a callback when all of a set of operations have been
   completed.  A CompletionNoticer will have it's noticeCompletion
   method called on it (with the argument you specify) when all of the
   elements of a set have indicated completion of an operation by
   being passed to the CompletionNoticerJoin as the argument to it's
   noticeCompletion method.<p>

   There are two ways to specify the set of objects whose completion
   is joined into the final notification: 1) pass in a Hashtable with
   the elements of the set as it's keys; 2) call addElement with each
   element of the set, followed by a call to enable().  If you did not
   specify an initial Hashtable, you must call enable() before you
   will receive notification.
*/
public class CompletionNoticerJoin implements CompletionNoticer 
{
    private CompletionNoticer myJoinedNoticer;
    private Object myJoinedNoticerArg;
    private Hashtable mySet;
    private boolean myEnabled;

    /**
       Create a new CompletionNoticerJoin specifing the arg to notify
       with, and an initial Hashtable of elements whose completion
       must be joined.  If the set is not null, this calls enable()
       for you.

       @param joinedNoticer the CompletionNoticer to be notified when
       all elements have reported completion.

       @param arg nullOk the argument that will be handed to joinedNoticer.

       @param set nullOk the initial set of objects to be joined.  If
       null, you will want to call addElement with the elements to be
       joined, and you must call enable() to receive your
       notification.
    */
    public CompletionNoticerJoin(CompletionNoticer joinedNoticer, Object arg, Hashtable set) {
        myJoinedNoticer = joinedNoticer;
        myJoinedNoticerArg = arg;
        if (set != null) {
            mySet = set ;
            enable();
        }
        else {
            mySet = new Hashtable();
        }
    }

    /**
       Create a new CompletionNoticerJoin.  joinedNoticer will be
       notified with noticeCompletion(null).  You must add any
       elements to join using addElement, and you must call enable()
       in order to receive your notification.

       @param joinedNoticer the CompletionNoticer to be notified when
       all elements have reported completion.
    */
    public CompletionNoticerJoin(CompletionNoticerJoin joinedNoticer) {
        this(joinedNoticer, null, null);
    }

    /**
       Extend the set of elements whose completion are required before
       we have completed.  Cannot be called after we've notified.  You
       may need to call enable() to receive your notification.  You
       can add more elements after enable()ing, and before
       notification, but it's probably not a good idea.

       @param element nullOk if non-null, you must arrange for
       this.noticeCompletion(element) to be called before the final
       notification will be triggered.
    */
    public void addElement(Object element) {
        if (mySet == null) {
            throw new Error("tried to add element after notification sent");
        }
        if (element != null) {
            mySet.put(element, element);
        }
    }

    /**
       Indicate that you're finished adding elements to the join.
       This must be called before a notification will be issued.  If
       you specify a non-null set when constructing the
       CompletionNoticerJoin, enable() will be called for you.
       Otherwise, you must call enable() yourself to receive your
       notification.
    */
    public void enable() {
        myEnabled = true;
        noticeCompletion(null);
    }

    /**
       Remove an element from the join.  When all elements have been
       removed, and enable() has been called, the final notification
       will be issued.  Only one notification will be issued.

       @param arg nullOk if non-null, the element that has completed.
    */
    public void noticeCompletion(Object arg) {
        if (arg != null && mySet != null) {
            mySet.remove(arg);
        }
        if (myEnabled && mySet != null && mySet.size() == 0 && myJoinedNoticer != null) {
            myJoinedNoticer.noticeCompletion(myJoinedNoticerArg);
            myJoinedNoticer = null;
            myJoinedNoticerArg = null;
            mySet = null;
        }
    }

    public String toString() {
        String ret = "CompletionNoticerJoin, still waiting on: " ;
        if (mySet == null) {
            ret += "nothing" ;
        }
        else {
            Enumeration en = mySet.keys();
            while (en.hasMoreElements()) {
                ret += " " + en.nextElement().toString() ;
            }
        }
        return ret;
    }
}
