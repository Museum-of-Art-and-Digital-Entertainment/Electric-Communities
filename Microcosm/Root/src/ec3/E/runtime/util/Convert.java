
/*
    Conversion Utilities
*/

package ec.util;

import java.util.Vector;
import java.util.Enumeration;

public class Convert {

/* Usefull functions for converting ints and longs into
   byte arrays and back.
*/
        public static byte[] convertToBytes(long inLong) {
        byte outv[] = new byte[8];
                outv[0] = (byte) ((inLong >>> 56) & 0xFF);
                outv[1] = (byte) ((inLong >>> 48) & 0xFF);
                outv[2] = (byte) ((inLong >>> 40) & 0xFF);
                outv[3] = (byte) ((inLong >>> 32) & 0xFF);
                outv[4] = (byte) ((inLong >>> 24) & 0xFF);
                outv[5] = (byte) ((inLong >>> 16) & 0xFF);
                outv[6] = (byte) ((inLong >>>  8) & 0xFF);
                outv[7] = (byte) (inLong & 0xFF);
                return(outv);
        }

        public static byte[] convertToBytes(int inInt) {
        byte outv[] = new byte[4];
                outv[0] = (byte) ((inInt >>> 24) & 0xFF);
                outv[1] = (byte) ((inInt >>> 16) & 0xFF);
                outv[2] = (byte) ((inInt >>> 8) & 0xFF);
                outv[3] = (byte) (inInt & 0xFF);
                return(outv);
        }

    public static long convertToLong(byte inv[]) {
    return (((long)inv[0] << 56) & ((long)0xFF << 56)) |
           (((long)inv[1] << 48) & ((long)0xFF << 48)) |
           (((long)inv[2] << 40) & ((long)0xFF << 40)) |
           (((long)inv[3] << 32) & ((long)0xFF << 32)) |
           (((long)inv[4] << 24) & ((long)0xFF << 24)) |
           (((long)inv[5] << 16) & ((long)0xFF << 16)) |
           (((long)inv[6] <<  8) & ((long)0xFF <<  8)) |
           (((long)inv[7] <<  0) & ((long)0xFF <<  0));

    }

    public static int convertToInt(byte inv[]) {
    return ((inv[0]<< 24) & (0xFF << 24)) |
           ((inv[1]<< 16) & (0xFF << 16)) |
           ((inv[2]<<  8) & (0xFF <<  8)) |
           ((inv[3]<<  0) & (0xFF <<  0));

    }

    public static byte[] StringToByteArray(String s) {
        char[] c = s.toCharArray();
        byte[] b = new byte[c.length];
        for (int i = 0; i < b.length; i++) {
            b[i] = (byte) c[i];
        }
        return b;
    }

    private static final char[] hexdigits = { '0', '1', '2', '3', '4',
        '5', '6', '7', '8', '9', 'A', 'B', 'C', 'D', 'E', 'F' };

    public static String ByteArrayToHexString(byte[] b) {
        char[] c = new char[b.length * 2];
        int j = 0;
        for (int i = 0; i < b.length; i++) {
            int nybble;
            nybble = (b[i] >> 4) & 0xF;
            c[j++] = hexdigits[nybble];
            nybble = b[i] & 0xF;
            c[j++] = hexdigits[nybble];
        }
        return new String(c);
    }

  // HexStringToByteArray currently assumes input is a HexString without check
    public static byte[] HexStringToByteArray(String s) {
        char[] c = s.toCharArray();
        byte[] b = new byte[c.length/2];
        int i,j;
        for (i = 0, j = 0; i < b.length; i++, j +=2) {
            // System.out.println("c[j],c[j+1] = " + c[j] + " " + c[j+1]);
            char m = (char)(Convert.HexCharToOffset(c[j]) & 0xF);
            char n = (char)(m << 4);
            char o = (char) (n & 0xF0);
            char r = (char)( Convert.HexCharToOffset(c[j+1]) & 0xF);
            char t = (char)( r & 0xF);
          b[i] = (byte)(o+t);
            // System.out.println("m = " + (int)m + " n =  " + (int)n + " o = " + (int)o
          //                                         + " r =  " + (int)r + " t = " + (int)t + " b[] = " + b[i]);
        }
        return b;
    }

    // HexCharToOffset currently assumes input is a HexChar without check
    // FIX caller must mask
    private static int HexCharToOffset(int c) {
        if (c <= '9')
            return (char)(c - '0');
        else
            return (char)(c - '0' - 7); // adjust to reflect character set ordering
    }

    public static String EscapeChar(String s, char c) {
          StringBuffer s_out = new StringBuffer(s.length() + 16);
          char tc;
          for(int i = 0; i < s.length(); i++) {
            tc = s.charAt(i);
            if (tc == '\\') {
              s_out.append('\\');
              s_out.append('\\');
            } else if (tc == c) {
              s_out.append('\\');
              s_out.append(c);
            } else {
              s_out.append(tc);
            }
          }
          return new String(s_out);
    }

        public static char[] expn_char (char[] in, int grow) {
        char[] tchar = new char[in.length+grow];

        System.arraycopy(in,0,tchar,0,in.length);
        return tchar;
        }
}
