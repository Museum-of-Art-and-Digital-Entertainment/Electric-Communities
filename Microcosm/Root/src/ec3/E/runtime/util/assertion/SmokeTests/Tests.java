import ec.util.assertion.*;
import java.util.*;

class Tests
{
  public static void main (String[] args)
  {
    new Tests().tests();
  }

  void tests()
  {
    loop: for (int test = 1; ; test++)
      {
        try
          {
            switch(test)
              {
              case 1: test(true); break;
              case 2: test(false); break;
              case 3: testNoString(true); break;
              case 4: testNoString(false); break;
              case 5: fail(); break;
              case 6: failNoString(); break;
              case 7: /* defunct */ break;
              case 8: /* defunct */ break;
              case 9: /* defunct */ break;
              case 10: /* defunct */ break;
              case 11: /* defunct */ break;
              case 12: /* defunct */ break;
              case 13: /* defunct */ break;
              case 14: test2(true); break;
              case 15: test3(true); break;
              case 16: test4(true); break;
              case 17: test5(true); break;
              case 18: test6(true); break;
              case 19: test7(true); break;
              case 20: test8(true); break;
              case 21: test2(false); break;
              case 22: test3(false); break;
              case 23: test4(false); break;
              case 24: test5(false); break;
              case 25: test6(false); break;
              case 26: test7(false); break;
              case 27: test8(false); break;
              default: break loop;
              }
          }
        catch (AssertionFailed e) { System.out.println(e.getMessage()); }
      }
    // This one is uncaught, to check the stack trace.
    // Note that this assumes all printlns are flushed before
    // assertion messages are printed to stderr.
    System.out.println("=== Finishing: Expect stack trace.");
    Assertion.fail();
  }

  void test(boolean b)
  {
    System.out.println("== test");
    Assertion.test(b, "Test assertion fires.");
    System.out.println("test: Assertion does not fire because b is " + b);
  }

  void testNoString(boolean b)
  {
    System.out.println("== testNoString");
    Assertion.test(b);
    System.out.println("testNoString: Assertion does not fire because b is " + b);
  }

  void test2(boolean b)
  {
    System.out.println("== test with 2 explanation args");
    Assertion.test(b, "Test assertion fires.", new Integer(2));
    System.out.println("test: Assertion does not fire because b is " + b);
  }

  void test3(boolean b)
  {
    System.out.println("== test with 3 explanation args");
    Assertion.test(b, "Test assertion fires.",
                   new Integer(2), new Integer(3));
    System.out.println("test: Assertion does not fire because b is " + b);
  }

  void test4(boolean b)
  {
    System.out.println("== test with 4 explanation args");
    Assertion.test(b, "Test assertion fires.", new Integer(2),
                   new Integer(3), new Integer(4));
    System.out.println("test: Assertion does not fire because b is " + b);
  }

  void test5(boolean b)
  {
    System.out.println("== test with 5 explanation args");
    Assertion.test(b, "Test assertion fires.", new Integer(2),
                   new Integer(3), new Integer(4), new Integer(5));
    System.out.println("test: Assertion does not fire because b is " + b);
  }

  void test6(boolean b)
  {
    System.out.println("== test with 6 explanation args");
    Assertion.test(b, "Test assertion fires.", new Integer(2),
                   new Integer(3), new Integer(4), new Vector(),
                   new Integer(6));
    System.out.println("test: Assertion does not fire because b is " + b);
  }

  void test7(boolean b)
  {
    System.out.println("== test with 7 explanation args");
    Assertion.test(b, "Test assertion fires.", new Integer(2),
                   new Integer(3), new Integer(4), new Vector(),
                   new Integer(6), new Long(7));
    System.out.println("test: Assertion does not fire because b is " + b);
  }

  void test8(boolean b)
  {
    System.out.println("== test with 8 explanation args");
    Assertion.test(b, "Test assertion fires.", new Integer(2),
                   new Integer(3), new Integer(4), new Vector(),
                   new Integer(6), new Long(7), new Float(8));
    System.out.println("test: Assertion does not fire because b is " + b);
  }

  void fail()
  {
    System.out.println("== fail");
    Assertion.fail("Test assertion fires.");
    System.out.println("==== FAILURE ===== fail: should not be reached.");
  }

  void failNoString()
  {
    System.out.println("== failNoString");
    Assertion.fail();
    System.out.println("==== FAILURE ===== failNoString: should not be reached.");
  }

}
