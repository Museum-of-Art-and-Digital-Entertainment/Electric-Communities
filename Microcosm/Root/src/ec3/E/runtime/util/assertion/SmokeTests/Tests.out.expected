== test
test: Assertion does not fire because b is true
== test
Internal program error: Test assertion fires.
== testNoString
testNoString: Assertion does not fire because b is true
== testNoString
Internal program error: Assertion failed.
== fail
Internal program error: Test assertion fires.
== failNoString
Internal program error: 'Unreachable' code was reached.
== test with 2 explanation args
test: Assertion does not fire because b is true
== test with 3 explanation args
test: Assertion does not fire because b is true
== test with 4 explanation args
test: Assertion does not fire because b is true
== test with 5 explanation args
test: Assertion does not fire because b is true
== test with 6 explanation args
test: Assertion does not fire because b is true
== test with 7 explanation args
test: Assertion does not fire because b is true
== test with 8 explanation args
test: Assertion does not fire because b is true
== test with 2 explanation args
Internal program error: Test assertion fires.2
== test with 3 explanation args
Internal program error: Test assertion fires.23
== test with 4 explanation args
Internal program error: Test assertion fires.234
== test with 5 explanation args
Internal program error: Test assertion fires.2345
== test with 6 explanation args
Internal program error: Test assertion fires.234[]6
== test with 7 explanation args
Internal program error: Test assertion fires.234[]67
== test with 8 explanation args
Internal program error: Test assertion fires.234[]678.0
=== Finishing: Expect stack trace.
ec.util.assertion.AssertionFailed: Internal program error: 'Unreachable' code was reached.
	at ec.util.assertion.Assertion.fail(Assertion.java:231)
	at Tests.tests(Tests.java:55)
	at Tests.main(Tests.java:8)
